### PR TITLE
Fix for `std::numeric_limits<std::_Signed128>::is_signed`

### PR DESCRIPTION
**Pull request by mtezych**
Hello,

I've noticed that the MSVC's STL provides `std::numeric_limits` template specialization
for the `std::_Signed128` (a type which models the `signed-integer-class` concept) slightly incorrectly,
that is the `std::numeric_limits<std::_Signed128>::is_signed` is `false` instead of `true`:

- https://godbolt.org/z/EcbE7Mrnq

This change fixes that issue and adds a couple of compile-time checks
verifying that the `std::numeric_limits<std::_Signed128>` is properly specialized.

### CODE REVIEW

File: stl/inc/__msvc_int128.hpp [hunk 1]
```diff
@@ -1407,8 +1407,9 @@ class numeric_limits<_Signed128> : public _Num_int_base {
         return 0;
     }
 
-    static constexpr int digits   = 127;
-    static constexpr int digits10 = 38;
+    static constexpr bool is_signed = true;
+    static constexpr int digits     = 127;
+    static constexpr int digits10   = 38;
```
Comment (CaseyCarter) [inline stl/inc/__msvc_int128.hpp:1412]: Let's follow the order in which these members are declared in the Standard:
```suggestion
    static constexpr int digits     = 127;
    static constexpr int digits10   = 38;
    static constexpr bool is_signed = true;
```

**Review by CaseyCarter — APPROVED** [review]
Thanks for fixing my mistake!

**Review by strega-nil-ms — APPROVED** [review]
Thanks for noticing!

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for fixing this bug, and congratulations on your first microsoft/STL commit! :tada: :heart_eyes_cat: :rocket:

This will ship in VS 2022 17.6 Preview 2.

**Pull request merged**